File: modules/core/src/main/java/com/ecfront/dew/core/entity/StatusEntity.java
package com.ecfront.dew.core.entity;

import javax.persistence.Column;
import javax.persistence.MappedSuperclass;

@MappedSuperclass
public abstract class StatusEntity extends IdEntity {

    @Column(nullable = false)
    protected Boolean enable;

    public Boolean getEnable() {
        return enable;
    }

    public void setEnable(Boolean enable) {
        this.enable = enable;
    }

}

File: components/auth/src/main/scala/com/tairanchina/csp/dew/auth/domain/StatusEntity.scala
package com.tairanchina.csp.dew.auth.domain

import javax.persistence.{Column, MappedSuperclass}

import scala.beans.BeanProperty

@MappedSuperclass
abstract class StatusEntity {

  @Column(nullable = false)
  @BeanProperty
  var enabled: Boolean = true

}